src/App.jsx

```diff
@@ -14,70 +15,66 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selected, setSelected] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const onClick = (product) => {
+    setSelected(selected === product
+      ? 'No goods selected'
+      : product);
+  };
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
+  return (
+    <main className="section container">
+      {selected !== 'No goods selected'
+        ? (
+          <h1 className="title is-flex is-align-items-center">
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={() => setSelected('No goods selected')}
+            />
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+            {`${selected} is selected`}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+          </h1>
+        )
+        : (
+          <h1 className="title is-flex is-align-items-center">
+            No goods selected
+          </h1>
+        )}
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              className={`${selected === good ? 'has-background-success-light' : ''}`}
+              data-cy="Good"
+              key={good}
             >
-              +
-            </button>
-          </td>
+              <td>
+                <button
+                  id={good}
+                  onClick={() => onClick(good)}
+                  data-cy={selected === good
+                    ? 'RemoveButton'
+                    : 'AddButton'}
+                  type="button"
+                  className={`button ${selected === good ? 'is-info' : ''}`}
+                >
+                  {selected === good ? '-' : '+'}
```

Comment:
make variable isSelected and use in all places.
`const isSelected = selected === good`